src/main/scala/com/microsoft/hyperspace/actions/CreateActionBase.scala

```diff
@@ -85,8 +85,9 @@ private[actions] abstract class CreateActionBase(dataManager: IndexDataManager)
               IndexLogEntry.schemaString(schema),
               numBuckets)),
           Content(
-            absolutePath.toString,
-            Seq(Directory("", indexFilesInfo(absolutePath), NoOpFingerprint()))),
+            root = absolutePath.getParent.toString,
+            directories = Seq(
+              Directory(absolutePath.getName, indexFilesInfo(absolutePath), NoOpFingerprint()))),
```

Comment:
I am actually fine leaving the `Directory.path` as "" to make it simple for now.

For example, if you have
```
/root/p=1/f1.txt
/root/p=2/f2.txt
```
We should theoretically store them as
```
root = "/root"
directories = Seq(Directory("p=1", Seq(FileInfo("f1"), _), Directory("p=2", Seq(FileInfo("f2"), _))
```
but it could get complicated to optimize when you have multiple partitions, so the following is also fine:
```
root = "/root"
directories = Seq(Directory("", Seq(FileInfo("p=1/f1"), _), Directory("", Seq(FileInfo("p=2/f2"), _))
```
as long as `full file path = "root/dir/filename"` holds true.

Reply:
yes that works for me. Let me change the code to this format.
Question @imback82 : is it fine to switch from directories to files while using indexes (e.g. in this [line](https://github.com/microsoft/hyperspace/blob/3a1dfa74ca369eedf8f19330f131dc21085a1c84/src/main/scala/com/microsoft/hyperspace/index/rules/FilterIndexRule.scala#L107) use individual files in `InMemoryFileIndex.rootPaths` field.

Reply:
we will eventually have to move here for `optimize`. cc @sezruby , @pirz

Reply:
> yes that works for me. Let me change the code to this format.

Btw, it was just a suggestion. If you can handle it in this PR, please go ahead


> Question @imback82 : is it fine to switch from directories to files while using indexes

I think it will be fine, but I would measure it once at least. Btw, we can also utilize `PathFilter` instead if needed.

Reply:
> > yes that works for me. Let me change the code to this format.
> 
> Btw, it was just a suggestion. If you can handle it in this PR, please go ahead
Oh thanks for clarification @imback82 . In that case, let's do the fix in this PR itself. I will ask @pirz and @sezruby if it works for them.
> 
> > Question @imback82 : is it fine to switch from directories to files while using indexes
> 
> I think it will be fine, but I would measure it once at least. Btw, we can also utilize `PathFilter` instead if needed (I am not sure which one will be better since we would already have the diff).
Sure, I will start some benchmarks before final merging of this. Thanks.

Reply:
cc @pirz , @sezruby , could you please have a general look at the code if it works with your changes?
(not looking for a full review, just a quick glance should suffice)

Reply:
Actually @apoorvedave1 has a better plan to revamp the `Content` structure. So we can close this PR for now. Thanks!

Reply:
@apoorvedave1 I think changing index directory structure won't affect #123.

> Question @imback82 : is it fine to switch from directories to files while using indexes

Could you let me know why do you need this?

BTW I think you could also use BucketUnion(#123) to read multiple root paths though it doesn't support outOrdering. What do you think?